### PR TITLE
Updated kotlin example

### PR DESCRIPTION
**Pull request by WeaponMan**
I updated to last eap version of kotlin, because 1.0.5-3 with kapt2 is unable to find AndroidManifest.
I added shared preferences example that is not working with kotlin.

### CODE REVIEW

**Comment by WonderCsabo** [comment]
@WeaponMan thanks for updating the sample. Can you remove the non-working Pref from this PR? I want to merge this, but we should not commit non-working example code. 

We want to send the non-working code to kotlin devs ofc.

**Comment by dodgex** [comment]
also i'm not sure if we want to use kotlin EAP in the sample project?

**Comment by WonderCsabo** [comment]
I agree, we should only use stable versions. Why did you update to EAP?
Again, it is ok in the test branch which we will show to Kotlin devs.

**Comment by WeaponMan** [comment]
I updated to EAP, because kapt2 was unable to find AndroidManifest file.
EAP version containts new kapt3 code that works mostly ok.
But it seams that generated code is not added to classpath of kapt gradle task.
I tried this without plugin kotlin-kapt which brings back the first version of kapt.
The build errors at generating stubs. (Unable to compile Cl class generated by kapt).

I will revert non working stuff and remove eap version from this PR tomorrow.

**Comment by WeaponMan** [comment]
Updated to 1.0.6 which was released on Dec 26. Totally missed that.

File: examples/kotlin/build.gradle [hunk 1]
```diff
@@ -1,10 +1,11 @@
 buildscript {
+    ext.kotlin_version = '1.0.6'
```
Comment (WonderCsabo) [inline examples/kotlin/build.gradle]: Add a new line after this.

File: examples/kotlin/src/main/kotlin/org/androidannotations/gradle/activity/HelloAndroidActivity.kt [hunk 2]
```diff
@@ -12,7 +12,7 @@ import org.androidannotations.gradle.R
 import java.util.Date
 
 @EActivity(R.layout.main)
-open public class HelloAndroidActivity : Activity() {
+open class HelloAndroidActivity : Activity() {
```
Comment (WonderCsabo) [inline examples/kotlin/src/main/kotlin/org/androidannotations/gradle/activity/HelloAndroidActivity.kt]: Why did you remove `public`?

**Comment by WonderCsabo** [comment]
@WeaponMan pls next time use a topic branch, not just `develop` as a target branch.

File: examples/kotlin/src/main/kotlin/org/androidannotations/gradle/activity/HelloAndroidActivity.kt (hunk 2 above)
Comment (WeaponMan) [inline examples/kotlin/src/main/kotlin/org/androidannotations/gradle/activity/HelloAndroidActivity.kt]: Classes with no access modifier are always public in kotlin. There is no need to type public.

Comment (WonderCsabo) [inline examples/kotlin/src/main/kotlin/org/androidannotations/gradle/activity/HelloAndroidActivity.kt]: Good to know, thx.

**Comment by WeaponMan** [comment]
Squashed into one commit.

File: examples/kotlin/build.gradle [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
+    
     repositories {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.0'
-        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.0.2'
+        classpath 'com.android.tools.build:gradle:2.2.3'
```
Comment (dodgex) [inline examples/kotlin/build.gradle]: when updating versions in this sample we should also update the versions in the gradle sample (and if necessary in the maven samples too).

**Comment by dodgex** [comment]
Thank you for your contribution @WeaponMan .

I added some additional comments. Unfortunately the PR is already merged.

@WonderCsabo could you have a look at those comments? i think at least for the updated versions in the other samples we should have another PR.

File: examples/kotlin/build.gradle [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
```
Comment (WonderCsabo) [inline examples/kotlin/build.gradle]: This is needed because of string interpolation.

Comment (dodgex) [inline examples/kotlin/build.gradle]: okay. do we want to be consistent and update the other lines too? if it is no noticeable performance drawback?

**Comment by WonderCsabo** [comment]
Yes, we should update all samples (include the maven one), to use latest and greatest plugin and dependency versions.

File: examples/kotlin/build.gradle (hunk 4 above)
Comment (WonderCsabo) [inline examples/kotlin/build.gradle]: I think we should use double quotes in all Gradle scripts for strings, yes. It looks better for consistency.

**Comment by dodgex** [comment]
Added new issue for updating the samples https://github.com/androidannotations/androidannotations/issues/1932
